modules/standard/BigInteger.chpl

```diff
@@ -4593,156 +4831,226 @@ module BigInteger {
+  proc bigint.cmp(x: real) : int {
+    const x_ = x : c_double;
     var   ret: c_int;
 
     if compiledForSingleLocale() {
-      ret = mpz_cmp_d(this.mpz, b_);
+      ret = mpz_cmp_d(this.mpz, x_);
 
     } else if this.localeId == chpl_nodeID {
-      ret = mpz_cmp_d(this.mpz, b_);
+      ret = mpz_cmp_d(this.mpz, x_);
 
     } else {
       const thisLoc = chpl_buildLocaleID(this.localeId, c_sublocid_any);
 
       on __primitive("chpl_on_locale_num", thisLoc) {
-        ret = mpz_cmp_d(this.mpz, b_);
+        ret = mpz_cmp_d(this.mpz, x_);
       }
     }
 
     return ret.safeCast(int);
   }
 
-  proc bigint.cmpabs(const ref b: bigint) : int {
+  pragma "last resort"
+  @deprecated("the argument name 'b' is deprecated - please use 'x' instead")
+  proc bigint.cmp(const ref b: bigint) : int do return this.cmp(b);
+
+  pragma "last resort"
+  @deprecated("the argument name 'b' is deprecated - please use 'x' instead")
+  proc bigint.cmp(b: int) : int do return this.cmp(b);
+
+  pragma "last resort"
+  @deprecated("the argument name 'b' is deprecated - please use 'x' instead")
+  proc bigint.cmp(b: uint) : int do return this.cmp(b);
+
+  pragma "last resort"
+  @deprecated("the argument name 'b' is deprecated - please use 'x' instead")
+  proc bigint.cmp(b: real) : int do return this.cmp(b);
+
+  /*
+     Compares the absolute value of ``this`` and the absolute value of ``x``.
+
+     :arg x: The number to compare against
+     :type x: :record:`bigint`, ``uint``, ``real``
+     :returns: Returns a positive value if ``abs(this)`` is greater than
+               ``abs(x)``, a negative value if ``abs(this)`` is less than
+               ``abs(x)``, or zero if they are equal.
+     :rtype: ``int``
+
+     .. seealso::
+        :proc:`GMP.mpz_cmpabs` and
+        `mpz_cmpabs <https://gmplib.org/manual/Integer-Comparisons#index-mpz_005fcmpabs>`_.
+  */
+  proc bigint.cmpabs(const ref x: bigint) : int {
     var ret: c_int;
 
     if compiledForSingleLocale() {
-      ret = mpz_cmpabs(this.mpz, b.mpz);
+      ret = mpz_cmpabs(this.mpz, x.mpz);
 
-    } else if this.localeId == chpl_nodeID && b.localeId == chpl_nodeID {
-      ret = mpz_cmpabs(this.mpz, b.mpz);
+    } else if this.localeId == chpl_nodeID && x.localeId == chpl_nodeID {
+      ret = mpz_cmpabs(this.mpz, x.mpz);
 
     } else {
       const thisLoc = chpl_buildLocaleID(this.localeId, c_sublocid_any);
 
       on __primitive("chpl_on_locale_num", thisLoc) {
-        var b_ = b;
+        var x_ = x;
 
-        ret = mpz_cmpabs(this.mpz, b_.mpz);
+        ret = mpz_cmpabs(this.mpz, x_.mpz);
       }
     }
 
     return ret.safeCast(int);
   }
 
-  proc bigint.cmpabs(b: uint) : int {
-    const b_ = b.safeCast(c_ulong);
+  /* See :proc:`~bigint.cmpabs` */
+  proc bigint.cmpabs(x: real) : int {
+    const x_ = x : c_double;
     var   ret: c_int;
 
     if compiledForSingleLocale() {
-      ret = mpz_cmpabs_ui(this.mpz, b_);
+      ret = mpz_cmpabs_d(this.mpz, x_);
```

Comment:
Is this `mpz_cmpabs_d` from `mpz_cmpabs_ui` change a case we had wrong before?

Reply:
There should be no change to this, it was correct before.

I think the way the diff is formatted is misleading, I had to double check several times.